Source/Common/FEXServerClient.cpp

```diff
@@ -143,7 +144,21 @@ namespace FEXServerClient {
   }
 
   fextl::string GetServerSocketName() {
-    return fextl::fmt::format("{}.FEXServer.Socket", ::geteuid());
+    constexpr std::string_view HARDCODED_SOCKET_PATH = "/FEX_Abstract_Socket";
+    if (FHU::Filesystem::Exists(HARDCODED_SOCKET_PATH.data())) {
+      // If this hardcoded filepath exists then use it.
+      // This is a required workaround for chroots changing users.
```

Comment:
What is the problem, and what is the actual workaround?

Reply:
Updated the comment.

Reply:
The PR description mentions something about environment variables. How are environment variables related to this? It's probably obvious, but not from the context here since the code below only deals with FEX's config.

Reply:
The rootfs scripts set the environment variable `FEX_SERVERSOCKETPATH` which typically gets inherited by the child processes: https://github.com/FEX-Emu/RootFS/blob/main/unbreak_chroot.sh#L120

Once an application in the chroot does a sudo/su, that environment variable stops getting inherited by default. Which means we can't rely on environment variables existing to inherit where FEXServer's abstract socket lives.

Reply:
For su/sudo specifically, could we configure the sudoers file to preserve environment variables instead?

If not, do we need the `FEX_SERVERSOCKETPATH` code at all anymore if this is what the rootfs scripts will now be using?

Reply:
We still need the environment variable for the FEXServer that is starting up.
I used su/sudo as an example, the apt/gnupg or whatever can drop the variable in a multitude of ways.

Reply:
> I used su/sudo as an example, the apt/gnupg or whatever can drop the variable in a multitude of ways.

Does this happen even for variables listed in `/etc/environment`? It's unclear what actual mechanism is triggering this, and if apt/gnupg also run into it, does it have anything to do with user switching at all?

Reply:
Looks like `/etc/environment` works around the issue today. Removed the hardcoded path, just made sure the environment variable works.